packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/useCreditCard.js

```diff
@@ -0,0 +1,439 @@
+ *
+ * @param {Boolean} props.updateButtonClicked boolean value which represents if a payment nonce request has been submitted
+ * @param {Function} props.onSuccess callback to invoke when the a payment nonce has been generated
+ * @param {Function} props.onReady callback to invoke when the braintree dropin component is ready
+ * @param {Function} props.onError callback to invoke when the braintree dropin component throws an error
+ * @param {Function} props.resetUpdateButtonClicked callback to reset the updateButtonClicked flag
+ * @param {DocumentNode} props.queries.getAllCountriesQuery query to fetch all countries data
+ * @param {DocumentNode} props.queries.getBillingAddressQuery query to fetch billing address from cache
+ * @param {DocumentNode} props.queries.getIsBillingAddressSameQuery query to fetch is billing address same checkbox value from cache
+ * @param {DocumentNode} props.queries.getPaymentNonceQuery query to fetch payment nonce saved in cache
+ * @param {DocumentNode} props.mutations.setBillingAddressMutation mutation to update billing address on the cart
+ * @param {DocumentNode} props.mutations.setCreditCardDetailsOnCartMutation mutation to update payment method and payment nonce on the cart
+ *
+ * @returns {
+ *   shouldRequestPaymentNonce: Boolean,
+ *   onPaymentError: Function,
+ *   onPaymentSuccess: Function,
+ *   onPaymentReady: Function,
+ *   isBillingAddressSame: Boolean,
+ *   countries: Object,
+ *   isDropinLoading: Boolean,
+ *   errors: Array<String>,
+ *   stepNumber: Number
+ * }
+ */
+export const useCreditCard = props => {
+    const {
+        onSuccess,
+        queries,
+        mutations,
+        onReady,
+        onError,
+        updateButtonClicked,
+        resetUpdateButtonClicked
+    } = props;
+    const {
+        getAllCountriesQuery,
+        getBillingAddressQuery,
+        getIsBillingAddressSameQuery,
+        getPaymentNonceQuery,
+        getShippingAddressQuery
+    } = queries;
+    const {
+        setBillingAddressMutation,
+        setCreditCardDetailsOnCartMutation
+    } = mutations;
+
+    /**
+     * Definitions
+     */
+
+    const [isDropinLoading, setDropinLoading] = useState(true);
+    const [shouldRequestPaymentNonce, setShouldRequestPaymentNonce] = useState(
+        false
+    );
+    /**
+     * `stepNumber` depicts the state of the process flow in credit card
+     * payment flow.
+     *
+     * `0` No call made yet
+     * `1` Billing address mutation intiated
+     * `2` Billing address mutation completed
+     * `3` Braintree nonce requsted
+     * `4` Braintree nonce received
+     * `5` Payment information mutation intiated
+     * `6` Payment information mutation completed
+     * `7` All mutations done
+     */
+    const [stepNumber, setStepNumber] = useState(0);
+
+    const client = useApolloClient();
+    const formState = useFormState();
+    const [{ cartId }] = useCartContext();
+
+    const { data: countriesData } = useQuery(getAllCountriesQuery);
```

Comment:
Why was this needed? I also can't find a `creditCard.js` component for this talon, so I'm thinking this might need re-named to match the component that uses it.

Reply:
My bad, `creditCardPaymentMethod.js` is the file you are looking for. I should have named it appropriately.